webdriver/tests/bidi/browsing_context/fragment_navigated/fragment_navigated.py

```diff
@@ -0,0 +1,302 @@
+    await bidi_session.browsing_context.navigate(context=new_tab["context"], url=target_url, wait="complete")
+    event = await on_entry
+
+    assert_navigation_info(event, {"context": new_tab["context"], "url": target_url})
+
+
+async def test_timestamp(bidi_session, current_time, subscribe_events, url, new_tab, wait_for_event):
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=url(EMPTY_PAGE), wait="complete"
+    )
+
+    await subscribe_events(events=[FRAGMENT_NAVIGATED_EVENT])
+
+    time_start = await current_time()
+
+    on_entry = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+    target_url = url(EMPTY_PAGE + '#foo')
+    await bidi_session.browsing_context.navigate(context=new_tab["context"], url=target_url, wait="complete")
+    event = await on_entry
+
+    time_end = await current_time()
+
+    assert_navigation_info(
+        event,
+        {"context": new_tab["context"], "timestamp": int_interval(time_start, time_end)}
+    )
+
+
+async def test_navigation_id(
+    bidi_session, new_tab, url, subscribe_events, wait_for_event
+):
+    await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=url(EMPTY_PAGE), wait="complete"
+    )
+
+    await subscribe_events([FRAGMENT_NAVIGATED_EVENT])
+
+    on_frame_navigated = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+
+    target_url = url(EMPTY_PAGE + '#foo');
+    result = await bidi_session.browsing_context.navigate(
+        context=new_tab["context"], url=target_url, wait="complete")
+
+    recursive_compare(
+        {
+            'context': new_tab["context"],
+            'navigation': result["navigation"],
+            'timestamp': any_int,
+            'url': target_url
+        },
+        await on_frame_navigated,
+    )
+
+
+async def test_url_with_base_tag(bidi_session, subscribe_events, inline, new_tab, wait_for_event):
+    url = inline("""<base href="/relative-path">""")
+    await bidi_session.browsing_context.navigate(context=new_tab["context"], url=url, wait="complete")
+
+    await subscribe_events(events=[FRAGMENT_NAVIGATED_EVENT])
+
+    on_frame_navigated = wait_for_event(FRAGMENT_NAVIGATED_EVENT)
+
+    target_url = url + '#foo';
+    await bidi_session.browsing_context.navigate(context=new_tab["context"], url=target_url, wait="complete")
+
+    recursive_compare(
+        {
+            'context': new_tab["context"],
+            'url': target_url
+        },
+        await on_frame_navigated,
+    )
+
+
+@pytest.mark.parametrize(
+    "context_kind",
+    ["top-level", "iframe"],
+    ids=[
+        "in a top-level context",
+        "in an iframe context",
+    ]
+)
+@pytest.mark.parametrize(
+    "navigation_kind",
+    ["browsing_context.navigate", "document.location", "history.pushState"],
+    ids=[
+        "navigation via browsing_context.navigate",
+        "navigation via document.location",
+        "navigation via history.pushState"
+    ]
+)
+@pytest.mark.parametrize(
```

Comment:
I feel with three levels of parameterization we clearly end-up with a matrix where a lot of tests are not needed and we waste time in running them.

As such I would really propose the following:

* There is no need to test everything for iframes. A single test here is really totally enough and that has to ensure that the contained `context` is the iframe's browsing context.
* Parameterization is fine when different values should be used for a given variable etc, but having huge code blocks behind doesn't really serve the purpose. As such lets really divide it up into three individual tests and each of them parameterize the hashes. Then we may not even need the ids given that a name like `test_document_location[#foo-#bar]` looks fine and is understandable.

Reply:
alright, I have split the tests. PTAL.